third_party/onnx-mlir.cmake

```diff
@@ -0,0 +1,19 @@
+include(ExternalProject)
+
+set(ONNX_MLIR_VERSION "v0.4.2.0")
+
+ExternalProject_Add(
+    onnx-mlir
+    PREFIX ${TTMLIR_TOOLCHAIN_DIR}
+    CMAKE_GENERATOR Ninja
+    CMAKE_ARGS
+    -DCMAKE_CXX_COMPILER=/usr/bin/c++
```

Comment:
Should this be `${CMAKE_CXX_COMPILER}`?

Reply:
Per the onnx build docs they suggest using this compiler on macOS, I think it's just clang, but in some kind of compatibility mode.  That said we should see test we really need it, might be sufficient to use regular clang.